import-service/services/import-file-parser/importFileParser.service.ts

```diff
@@ -9,21 +11,37 @@ type ImportFileParserServiceParams = {
 
 export const importFileParserService = async (
   params: ImportFileParserServiceParams
-) => {
+): Promise<Product[]> => {
   try {
     return new Promise(async (resolve, reject) => {
+      const result: Product[] = [];
       const { bucketName, key } = params;
       const s3 = new AwsSdkS3Service({ bucketName });
       const s3Object = await s3.getObject({ Key: key });
-      console.log({ s3Object });
+      console.log(
+        '🚀 ~ file: importFileParser.service.ts ~ line 19 ~ s3Object',
+        s3Object
+      );
       const rStream = s3Object.createReadStream();
       rStream
```

Comment:
Look for `utils promisify pipeline` to make work with streaming more effective and readable.
After that, you could use stream.Transformer to avoid usage of .on('data')